app/src/main/java/com/example/whattowear/DashboardActivity.java

```diff
@@ -26,4 +63,189 @@ public void onClick(View v) {
             }
         });
     }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        // request permissions if no permission given (can be moved to a button click or anywhere)
+        if (!hasLocationPermissions()) {
+            requestLocationPermissions();
+        }
+        Log.i(TAG, "Finished asking permissions");
+
+        updateWeather();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        // Forward results to EasyPermissions
+        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);
+    }
+
+    /**
+     * Used to check if location permissions have been granted
+     * @return  whether the application has locations services permissions
+     */
+    private boolean hasLocationPermissions() {
+        return EasyPermissions.hasPermissions(this, Manifest.permission.ACCESS_FINE_LOCATION);
+    }
+
+    /**
+     * Used to request location permissions
+     */
+    private void requestLocationPermissions() {
+        EasyPermissions.requestPermissions(
+                this,
+                "This application requires location services to auto detect your location.",
+                PERMISSIONS_REQUEST_CODE,
+                Manifest.permission.ACCESS_FINE_LOCATION);
+    }
+
+    @Override
+    public void onPermissionsGranted(int requestCode, @NonNull List<String> perms) {
+        Toast.makeText(this, "Location services permission granted!", Toast.LENGTH_SHORT).show();
+
+        updateWeather();
+    }
+
+    @Override
+    public void onPermissionsDenied(int requestCode, @NonNull List<String> perms) {
+        Toast.makeText(this, "Location services permission denied!", Toast.LENGTH_SHORT).show();
+
+        // Check whether the user denied any permissions and checked "NEVER ASK AGAIN."
+        // This will display a dialog directing them to enable the permission in app settings.
```

Comment:
I planed to deny usage of the app entirely, and redirect towards the settings page where the user can change permissions, since most of the features are based on their location. This does seem extreme though, and I could change it to still be usable with the user having to manually enter their location, but I think this is similar to the option where the app is granted access to the user's location just once. What do you think?

Reply:
Sounds good. Maybe we can still let them type their location instead of locating them directly? Feel free to add a TODO here, I may ask for more than what this app is needed.